hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3AUtils.java

```diff
@@ -254,7 +254,7 @@ public static IOException translateException(@Nullable String operation,
       case 404:
         if (isUnknownBucket(ase)) {
           // this is a missing bucket
-          ioe = new UnknownStoreException(path, ase);
+          ioe = new UnknownStoreException(message, ase);
```

Comment:
Should we make UnknownStoreException a subclass of PathIOE and supply the path in the constructor there